user/views.py

```diff
@@ -12,6 +15,72 @@ def check_login(request):
 
 
 def home(request):
-    if request.user.is_authenticated:        
+    if request.user.is_authenticated:
         return render(request,'tweet/home.html')
-    return redirect('index')
\ No newline at end of file
+    return redirect('index')
+
+def signup(request):
+    if request.user.is_authenticated:
+        return redirect('index')
+    else:
+        if request.method == 'GET':
+            return redirect('index')
+        if request.method == 'POST':
+            email = request.POST.get('email')
+            if not email:
+                messages.warning(request, "Email required")
+                return redirect('index')
+            if User.objects.filter(email=email):
+                messages.warning(request, "User already exists with this email")
+                return redirect('index')
+
+            full_name = request.POST.get('name')
+            full_username = "".join(full_name.split())
+            user_name = full_username
+
+            while User.objects.filter(username=user_name).exists():
+                random_number = random.randint(1, 9999)
+                user_name = f"{full_username}{random_number}"
+            User.objects.create(
+                password = make_password(request.POST.get('password')),
+                username = user_name,
+                first_name = full_name,
+                email=email
+            )
+            messages.success(request, "Login to continue")
+            return redirect('index')
+
+
+
+
+
+def signin(request):
+    if request.user.is_authenticated:
+        return redirect('index')
+    else:
```

Comment:
Also here see above

Reply:
minor change is done but need to discuss what if to run two functions in a single if this can be done or not in a single line as signin have 2-3 functions to check existance, match password